cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionMetadata.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import com.google.common.base.Objects;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Contains metadata associated with a particular {@link Partition} of a {@link @PartitionedFileSet}
+ */
+public class PartitionMetadata implements Iterable<Map.Entry<String, String>> {
+  private final Map<String, String> metadata;
+
+  public PartitionMetadata(Map<String, String> metadata) {
+    this.metadata = Collections.unmodifiableMap(new HashMap<>(metadata));
+  }
+
+  public String get(String key) {
```

Comment:
This method should be `@Nullable`. Also missing javadoc.

Also, do we allow `null` value in the map? If not, then the corresponding set/add/put method should check and bounce on null value.